public/app/core/components/UserEdit/UserProfile.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { ChangeEvent, PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import { getBackendSrv } from '@grafana/runtime';
+import { OrgUser, StoreState } from 'app/types';
+import { updateUser } from './state/actions';
+import { Input, Button, FormLabel } from '@grafana/ui';
+
+export interface Props {
+  adminMode?: boolean;
+  userId?: number;
+  user: OrgUser;
+  updateUser: typeof updateUser;
+}
+export interface State {
+  mode: string;
+  name: string;
+  email: string;
+  login: string;
+  password: string;
+}
+
+export class UserProfile extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { user, userId, adminMode } = props;
+    const { name, email, login } = user;
+    let mode = '';
+
+    if (adminMode && userId) {
+      mode = 'edit';
+    } else if (adminMode) {
+      mode = 'new';
+    } else {
+      mode = 'profile';
+    }
+
+    this.state = {
+      mode,
+      name,
+      email,
+      login,
+      password: '',
+    };
+  }
+
+  pageHeader() {
+    const { mode } = this.state;
+    let header: string;
+
+    switch (mode) {
+      case 'edit':
+        header = 'Edit User';
+        break;
+      case 'new':
+        header = 'Add New User';
+        break;
+      case 'new':
+        header = 'Edit Profile';
+        break;
+    }
+
+    return header;
+  }
+
+  onNameChange = (name: string) => {
+    this.setState({ name });
+  };
+
+  onEmailChange = (email: string) => {
+    this.setState({ email });
+  };
+
+  onLoginChange = (login: string) => {
+    this.setState({ login });
+  };
+
+  onPasswordChange = (password: string) => {
+    this.setState({ password });
+  };
+
+  async onCreateUser() {
```

Comment:
please move the create form to it's own component, Create & edit should not be done in the same component, we want to very soon have differently designed forms & UX for these.